### PR TITLE
Improve view pan/zoom controls

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Add 'zoom extents' and 'zoom to image' (a.k.a. 'reset view') actions to the NOAA GUI. Add ability to unlink views (so that pan/zoom are not synchronized), and always do so when no transform is loaded.

### CODE REVIEW

**Review by mattdawkins — APPROVED** [review]

File: sealtk/noaa/gui/Window.cpp [hunk 1]
```diff
@@ -206,9 +209,6 @@ class WindowPrivate
 
   QPointer<QShortcut> cancelToolShortcut;
 
-  float zoom = 1.0f;
-  QPointF center{0.0f, 0.0f};
```
Comment (russelldj) [inline sealtk/noaa/gui/Window.cpp:210]: This looks like an odd default for the center of an image.

File: sealtk/noaa/gui/Window.cpp [hunk 2]
```diff
@@ -602,17 +577,35 @@ void WindowPrivate::createWindow(
...
 
-  QObject::connect(q, &Window::zoomChanged,
-                   data->player, &sg::Player::setZoom);
-  QObject::connect(data->player, &sg::Player::zoomChanged,
-                   q, &Window::setZoom);
-  data->player->setZoom(q->zoom());
+  QObject::connect(
+    data->player, &sg::Player::zoomChanged, q,
+    [this, player = data->player](float zoom){
+      if (this->ui.actionViewLink->isChecked() && player->hasTransform())
```
Comment (russelldj) [inline sealtk/noaa/gui/Window.cpp:583]: If this does what I think it does and is synchronizing the zoom across multiple viewpoints, perhaps `actionLinkViews` would be *slightly* clearer.

File: sealtk/noaa/gui/Window.cpp [hunk 3]
```diff
@@ -1071,6 +1068,111 @@ QModelIndex WindowPrivate::modelIndex(
...
+      {
+        auto const windowSize = QSizeF{w->player->effectiveImageSize()};
+        auto const windowRect = QRectF{{0, 0}, windowSize};
+        w->player->setCenter(windowExtents.center() - windowRect.center());
+
+        if (types != sg::ImageExtents)
+        {
+          auto const adjust =
+            qMax(10.0, 0.1 * qMax(windowExtents.width(),
+                                  windowExtents.height()));
```
Comment (russelldj) [inline sealtk/noaa/gui/Window.cpp:1147]: Perhaps slightly confusing what the 10.0 represents and the overall goal. Is this a cap on zooming speed?

**Review by russelldj — APPROVED** [review]
Everything looked good to me. I made a few stylistic suggestions but I don't feel any of them are required. I didn't thoroughly review the math of the transforms, nor have I ever actual built `seal-tk`. If either of those things would be useful, I can certainly do them.

File: sealtk/noaa/gui/Window.cpp (hunk 2 above)
Comment (mwoehlke-kitware) [reply sealtk/noaa/gui/Window.cpp:583]: Huh... I was going to mutter something about hierarchical naming, but nothing else is using that convention. Changed; thanks!

File: sealtk/noaa/gui/Window.cpp (hunk 3 above)
Comment (mwoehlke-kitware) [reply sealtk/noaa/gui/Window.cpp:1147]: No. As above, this is an adjustment (padding) that is applied to the extents prior to fitting them in the view, which is 10 units¹, 10% of the image width, or 10% of the image height, whichever is largest.

(Due to possible homographies, I believe "unit" is effectively "EO image pixel"... as opposed to, say, image pixels in the view being zoomed. Definitely not *screen* pixels.)

**Pull request merged**